src/Uno.UI/UI/Xaml/DependencyPropertyDetailsCollection.cs

```diff
@@ -131,20 +131,13 @@ public DependencyPropertyDetails FindPropertyDetails(DependencyProperty property
 			// https://stackoverflow.com/a/17095534/26346
 			var isInRange = (uint)entryIndex <= (_maxId - _minId);
 
-			GetPropertyInheritanceConfiguration(
-				property: property,
-				hasInherits: out var hasInherits,
-				hasValueInherits: out var hasValueInherits,
-				hasValueDoesNotInherit: out var hasValueDoesNotInherits
-			);
```

Comment:
Previously, at this point, `GetPropertyInheritanceConfiguration` could have already calculated property metadata.

Then we create a DependencyPropertyDetails that has a null `_metadata` field that will be calculated once again then cached.

So in short, we will be calculating property metadata twice.

By moving GetPropertyInheritanceConfiguration to DependencyPropertyDetails, it simplifies the constructor and also guarantees that metadata is calculated once and cached in DependencyPropertyDetails. It's not very significant as it's a quite fast dictionary lookup, but many of such small improvements can build up. And I also think it simplifies the code a little bit anyway.

Reply:
Makes sense! Nice find!